customer/structures.go

```diff
@@ -78,11 +80,307 @@ type Configuration struct {
 	AllowedDomains []string                     `json:"allowed_domains,omitempty"`
 	Integrations   map[string]map[string]string `json:"integrations"`
 	Properties     struct {
-		Group   objects.Properties `json:"group"`
-		License objects.Properties `json:"license"`
+		Group   Properties `json:"group"`
+		License Properties `json:"license"`
 	} `json:"properties"`
 }
 
+// User represents base of both Customer and Agent
+//
+// To get speficic user type's structure, call Agent() or Customer() (based on Type value).
+type User struct {
+	ID             string    `json:"id"`
+	Type           string    `json:"type"`
+	Name           string    `json:"name"`
+	Avatar         string    `json:"avatar"`
+	Email          string    `json:"email"`
+	Present        bool      `json:"present"`
+	EventsSeenUpTo time.Time `json:"events_seen_up_to"`
+	userSpecific
+}
+
+type userSpecific struct {
+	RoutingStatus              json.RawMessage `json:"routing_status"`
+	LastVisit                  json.RawMessage `json:"last_visit"`
+	Statistics                 json.RawMessage `json:"statistics"`
+	AgentLastEventCreatedAt    json.RawMessage `json:"agent_last_event_created_at"`
+	CustomerLastEventCreatedAt json.RawMessage `json:"customer_last_event_created_at"`
+	SessionFields              json.RawMessage `json:"session_fields"`
+	Followed                   json.RawMessage `json:"followed"`
+	Online                     json.RawMessage `json:"online"`
+	State                      json.RawMessage `json:"state"`
+	GroupIDs                   json.RawMessage `json:"group_ids"`
+	EmailVerified              json.RawMessage `json:"email_verified"`
+	CreatedAt                  json.RawMessage `json:"created_at"`
```

Comment:
the majority of the fields don't exist in customer-api; I know it wasn't reflected earlier because these were common structs, but now we have a chance to define structs more closely